panic: unaligned 64-bit atomic operation

### Current Behavior

I encountered a runtime panic when running the hostd Docker container on my Raspberry Pi (aarch64).

Here's a little snippet of the log:

```
goroutine 51 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Store64(0x2c9010c, 0x3e)
	runtime/internal/atomic/atomic_arm.s:291 +0x14
go.sia.tech/hostd/wallet.(*SingleAddressWallet).ProcessConsensusChange(0x2c900e0, {{0xc4, 0x5, 0x69, 0x84, 0xbf, 0x96, 0x84, 0x32, 0xaf, ...}, ...})
	go.sia.tech/hostd/wallet/wallet.go:698 +0xba4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe.func2(0x2d28180)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:212 +0x19c
gitlab.com/NebulousLabs/bolt.(*DB).View(0x29003c0, 0x284ad2c)
	gitlab.com/NebulousLabs/bolt@v1.4.4/db.go:730 +0xa4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe(0x28c8a80, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:200 +0x33c
go.sia.tech/siad/modules/consensus.(*ConsensusSet).ConsensusSetSubscribe(0x28c8a80, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:256 +0xec
go.sia.tech/hostd/chain.(*Manager).Subscribe(0x2dba000, {0x10e9308, 0x2c900e0}, {0xef, 0xea, 0xf7, 0xe9, 0x3b, 0x5f, 0xf4, ...}, ...)
	go.sia.tech/hostd/chain/manager.go:133 +0x5c
go.sia.tech/hostd/wallet.NewSingleAddressWallet.func1()
	go.sia.tech/hostd/wallet/wallet.go:767 +0x7c
created by go.sia.tech/hostd/wallet.NewSingleAddressWallet
	go.sia.tech/hostd/wallet/wallet.go:765 +0x720
2023/06/18 11:15:34 hostd v0.1.0-alpha.3-32-g0be24c2
2023/06/18 11:15:34 Network Zen Testnet
{"level":"info","ts":1687086934.2192411,"logger":"stdlog","caller":"hostd/main.go:50","msg":"Using HOSTD_ZEN_API_PASSWORD environment variable."}
{"level":"info","ts":1687086934.220021,"logger":"stdlog","caller":"hostd/main.go:69","msg":"Using HOSTD_ZEN_SEED environment variable."}
panic: unaligned 64-bit atomic operation
```

### Expected Behavior

Work? 📈 😅

### Steps to Reproduce

Just build from the scratch and try to start the container 🤷‍♂️

```
git clone https://github.com/SiaFoundation/hostd.git && cd hostd

docker build -t zen_hostd -f docker/Dockerfile.testnet .  

docker run -d --name hostd -e HOSTD_ZEN_API_PASSWORD='PASS' -e HOSTD_ZEN_SEED='SEED' -p localhost:9880:9880/tcp -p :9881:9881/tcp zen_hostd
```

### Version

hostd v0.1.0-alpha.3-32-g0be24c2

### What operating system did the problem occur on (e.g. Ubuntu 22.04, macOS 12.0, Windows 11)?

Raspberry Pi OS | aarch64

### Anything else?

_No response_

Yeap, absolutely. In addition, the same Raspberry Pi is also carrying a renterd container, which works like clockwork.
I'm very impressed that you managed to do that. I spent a few days trying to exclude all the reasons why this might be happening. I can see the container starting up and hostd launching for about 5 seconds and if I take action quickly I'll be lucky enough to even be able to log in and enter the UI but then it crash. 

<img width="282" alt="Screenshot 2023-06-19 в 19 58 04" src="https://github.com/SiaFoundation/hostd/assets/17514297/d49f818d-3c78-4574-8c65-ddc90b339c19">

For the sake of the experiment's purity, I even tried running hostd on mainnet, although I understood that the result would likely be the same, and indeed it was.

```
2023/06/18 12:56:53 hostd v0.1.0-alpha.3-32-g0be24c2
2023/06/18 12:56:53 Network Mainnet
{"level":"info","ts":1687093013.470032,"logger":"stdlog","caller":"hostd/main.go:50","msg":"Using HOSTD_API_PASSWORD environment variable."}
{"level":"info","ts":1687093013.4703264,"logger":"stdlog","caller":"hostd/main.go:69","msg":"Using HOSTD_SEED environment variable."}
panic: unaligned 64-bit atomic operation

goroutine 69 [running]:
runtime/internal/atomic.panicUnaligned()
	runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Store64(0x350e56c, 0x0)
	runtime/internal/atomic/atomic_arm.s:291 +0x14
go.sia.tech/hostd/wallet.(*SingleAddressWallet).ProcessConsensusChange(0x350e540, {{0x89, 0xea, 0xd7, 0xf6, 0x71, 0x80, 0xa2, 0x8f, 0x5b, ...}, ...})
	go.sia.tech/hostd/wallet/wallet.go:698 +0xba4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe.func2(0x30c6600)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:212 +0x19c
gitlab.com/NebulousLabs/bolt.(*DB).View(0x31143c0, 0x30f3d2c)
	gitlab.com/NebulousLabs/bolt@v1.4.4/db.go:730 +0xa4
go.sia.tech/siad/modules/consensus.(*ConsensusSet).managedInitializeSubscribe(0x30d0a80, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:200 +0x33c
go.sia.tech/siad/modules/consensus.(*ConsensusSet).ConsensusSetSubscribe(0x30d0a80, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/siad@v1.5.10-0.20230228235644-3059c0b930ca/modules/consensus/subscribe.go:256 +0xec
go.sia.tech/hostd/chain.(*Manager).Subscribe(0x3194000, {0x10edae8, 0x350e540}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...)
	go.sia.tech/hostd/chain/manager.go:133 +0x5c
go.sia.tech/hostd/wallet.NewSingleAddressWallet.func1()
	go.sia.tech/hostd/wallet/wallet.go:767 +0x7c
created by go.sia.tech/hostd/wallet.NewSingleAddressWallet
	go.sia.tech/hostd/wallet/wallet.go:765 +0x720

```

--- a/host/contracts/manager.go
+++ b/host/contracts/manager.go
@@ -64,6 +64,8 @@ type (
 
 	// A ContractManager manages contracts' lifecycle
 	ContractManager struct {
+		blockHeight uint64 // ensure 64-bit alignment on 32-bit systems
+
 		store ContractStore
 		tg    *threadgroup.ThreadGroup
 		log   *zap.Logger
@@ -73,8 +75,6 @@ type (
 		chain   ChainManager
 		tpool   TransactionPool
 		wallet  Wallet
-
-		blockHeight uint64
 
 		processQueue chan uint64 // signals that the contract manager should process actions for a given block height
 

--- a/host/storage/storage.go
+++ b/host/storage/storage.go
@@ -64,6 +64,9 @@ type (
 
 	// A VolumeManager manages storage using local volumes.
 	VolumeManager struct {
+		cacheHits   uint64 // ensure 64-bit alignment on 32-bit systems
+		cacheMisses uint64
+
 		a        Alerts
 		vs       VolumeStore
 		cm       ChainManager
@@ -77,8 +80,6 @@ type (
 		// changedVolumes tracks volumes that need to be fsynced
 		changedVolumes map[int]bool
 		cache          *lru.Cache[types.Hash256, *[rhpv2.SectorSize]byte] // Added cache
-		cacheHits      uint64
-		cacheMisses    uint64
 	}
 )
 

--- a/wallet/wallet.go
+++ b/wallet/wallet.go
@@ -66,9 +66,10 @@ type (
 	// A SingleAddressWallet is a hot wallet that manages the outputs controlled by
 	// a single address.
 	SingleAddressWallet struct {
-		priv       types.PrivateKey
-		addr       types.Address
-		scanHeight uint64
+		scanHeight uint64 // ensure 64-bit alignment on 32-bit systems
+
+		priv types.PrivateKey
+		addr types.Address
 
 		cm    ChainManager
 		store SingleAddressStore

